C make error testing

JohnnyH-Dev opened the pull request:
<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

JohnnyH-Dev committed:
Delete build-and-publish-debug.yml

JohnnyH-Dev committed:
Delete build-and-publish.yml

JohnnyH-Dev committed:
Delete build-assets.yml

JohnnyH-Dev committed:
Delete build-containers-pr.yml

JohnnyH-Dev committed:
Delete build-containers-target-branch.yml

JohnnyH-Dev committed:
Delete build-test-eclwatch.yml

JohnnyH-Dev committed:
Delete build-vcpkg.yml

JohnnyH-Dev committed:
Delete codeql-eclwatch.yml

JohnnyH-Dev committed:
Delete deploy-docs.yml

JohnnyH-Dev committed:
Delete jirabot.yml

JohnnyH-Dev committed:
Delete nightly-publish.yml

JohnnyH-Dev committed:
Delete test-eclhelper.yml

JohnnyH-Dev committed:
Delete test-helm.yml

JohnnyH-Dev committed:
Delete timeoutcmd

JohnnyH-Dev committed:
Add files via upload

JohnnyH-Dev committed:
Delete smoketest.yml

JohnnyH-Dev committed:
Add files via upload

JohnnyH-Dev committed:
Delete smoketest.yml

JohnnyH-Dev committed:
Add files via upload

JohnnyH-Dev committed:
Update johnnytest4.sh

Updated the directory within the script to specifically look at .log files within the .github/workflows folder

JohnnyH-Dev committed:
Update smoketest.yml

Updated smoketest.yml file to atttemp to fix the error where it cannot find the johnnytest4.sh file, causing github actions to stop

JohnnyH-Dev committed:
Update smoketest.yml

Fixed yml syntax error

AttilaVamos commented on .github/workflows/smoketest.yml:
```diff
@@ -290,6 +290,12 @@ jobs:
 
+    - name: Run error handling script
+      run: |
+        chmod +x ./.github/workflows/johnnytest4.sh
```
In GitHub runner the directory structure is a bit different than as is in BM/VM. So, to use your script it is better to copy into the current directory:
```
cp -v ${{ github.workspace }}/src/.github/workflows/johnnytest4.sh .
```
then do the other things.

AttilaVamos commented on .github/workflows/johnnytest4.sh:
```diff
@@ -0,0 +1,70 @@
+output_file="CMake_Error_Log.txt"
+
+# Clean the output file first
+echo "" > $output_file
```
That seems ok, but if you rerun the CMake, next time this overwrite the previous log.
You may considering some things like:
- do we need the previous content of CMake_Erro_Log.txt is we re-run CMake?
- if yes then how to keep the previous content? (maybe adding a timestamp to the file name using date command?)
- if it is a log file, we usually using then '.log' as the extension.

AttilaVamos commented on .github/workflows/johnnytest4.sh:
```diff
@@ -0,0 +1,70 @@
+echo "" > $output_file
+
+# specify the directory
+directory="./.github/workflows"
```
It doesn't work in GitHub action environment. If you want to check where are you in directory tree then use pwd command or see my comment for your changes with smoketest.yaml.

AttilaVamos commented on .github/workflows/johnnytest4.sh:
```diff
@@ -0,0 +1,70 @@
+
+  # Load error message into a variable
+  error_message=$(awk '/CMake Error/{flag=1}/^$/{flag=0}flag' $input_file)
+  
```
It works, but if you should do same thing twice, you may considering to avoid it if it possible.
In this case, you can store the first awk result in a variable, then echo that variable into the $output_file. That means: store the awk result into error_message variable, then echo it into the $output_file

AttilaVamos commented on .github/workflows/johnnytest4.sh:
```diff
@@ -0,0 +1,70 @@
+    echo "No CMake errors detected." >> $output_file
+    continue
+  fi
+
```
Moreover I may put the echo $error_message >> $output_file into an ELSE branchlike this:
```
if [[ -z "$error_message" ]]; then
    echo "No CMake errors detected." >> $output_file
    continue
else
    echo $error_message >> $output_file
fi
```

AttilaVamos commented on .github/workflows/johnnytest4.sh:
```diff
@@ -0,0 +1,70 @@
+
+# Check error message for known issues
+# Check error message for known issues
+if [[ $error_message == *"add_subdirectory"* ]]; then
```
I didn't saw this condition before. Usually we are using '=~' operator.

AttilaVamos commented on .github/workflows/johnnytest4.sh:
```diff
@@ -0,0 +1,70 @@
+  echo "    - HPCC-Platform-build/CMakeFiles/CMakeError.log" >> $output_file
+elif [[ $error_message == *"Failed to download"* ]]; then
+  echo "  - Capture" >> $output_file
+  echo "sudo apt-get install -y bison flex build-essential binutils-dev curl lsb-release libcppunit-dev python3-dev default-jdk r-base-dev r-cran-rcpp r-cran-rinside r-cran-inline pkg-config libtool autotools-dev automake git cmake" >> $output_file
```
It is not a good practice to put this kind of information into your script. That means you hardwired it, but it is possible next week, month, year it will change. So If you want to report this kind of information you need to dig it up from log.

JohnnyH-Dev committed:
Update smoketest.yml

Updated the Run Error Handling Script directory so that it copies the shell script from the location within the repository to the current working directory for the runner.

JohnnyH-Dev committed:
Update johnnytest4.sh

Updated the script based off of Attila's comments

JohnnyH-Dev committed:
Update smoketest.yml

JohnnyH-Dev committed:
Update smoketest.yml

JohnnyH-Dev committed:
Update smoketest.yml

JohnnyH-Dev committed:
Update johnnytest4.sh

JohnnyH-Dev committed:
Update smoketest.yml

JohnnyH-Dev committed:
Add files via upload
